grpc-gcp/src/main/java/com/google/grpc/gcp/GcpManagedChannelOptions.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.grpc.gcp;
+
+import io.opencensus.metrics.LabelKey;
+import io.opencensus.metrics.LabelValue;
+import io.opencensus.metrics.MetricRegistry;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+/** Options for the {@link GcpManagedChannel}. */
+public class GcpManagedChannelOptions {
```

Comment:
Why do we need an extra layer of GcpManagedChannelOptions? I think MetricsOptions is the only object in it? What else do you plan to put in this GcpManagedChannelOptions?
What's the difference between what is configured in GcpManagedChannelBuilder (like pool size, apiconfig)?

Reply:
I hesitated to use apiconfig for the metrics options because we need to pass an instance to the MetricRegistry to report metrics to. It also feels like apiconfig is only for configuring the location of the affinity key in the RPCs and BIND/BOND/UNBIND semantic of RPCs. Maybe we should move maxSize and low watermark options to this new GcpManagedChannelOptions eventually? WDYT?
Yeah, I plan to add GcpResiliencyOptions there too. This will configure which resiliency improvements are enabled and their fine-tuned parameters.

Reply:
I agree that we should not use apiConfig for metrics options..
But can we just add a new property like MetricsOptions to GcpManagedChannelBuilder? I think GcpManagedChannelOptions and GcpManagedChannelBuilder kind of serves the same functionalities.

Reply:
Yeah, I agree the GcpManagedChannelOptions/Builder with a single thing inside looks a little awkward here, but I will add GcpResiliencyOptions alongside GcpMetricsOptions in the very next PR. And I want to have a different groups of options under the umbrela of GcpManagedChannelOptions. Does this plan sounds good?